sdk/client.py

```diff
@@ -103,32 +106,35 @@ def _request(self, method: str, url: str, payload: Dict = None):
         return data
 
     def _send_batch(
-        self, batch_data: Dict
+        self, batch_data: Dict[str, Any]
     ) -> Tuple[Dict[str, Any], List[Dict[str, Any]]]:
         batch_data.update({"group_id": self.group_id})
         batch_data = self._request(
             "POST",
             f"{self.endpoints.core}/api/v1/batches",
             batch_data,
         )["data"]
-        jobs_data = batch_data.pop("jobs", [])
+        jobs_data = batch_data.pop("jobs", {})
```

Comment:
Why replacing the list with a dict here?

Reply:
Interesting that mypy wasn't complaining about this

Reply:
I think it doesn't manage to understand the pop method. I tried for a time (and it might be still left in some places) to annotate the return type of what's `pop`ed. In that case it does work. But it also makes reading the code harder...

You'll see in below that in a different method here the default values of this was swapped.